tool/tsh/tsh.go

```diff
@@ -3132,11 +3136,33 @@ func makeClientForProxy(cf *CLIConf, proxy string, useProfileLogin bool) (*clien
 	// addresses from the proxy (this is what Ping does).
 	if cf.IdentityFileIn != "" {
 		log.Debug("Pinging the proxy to fetch listening addresses for non-web ports.")
-		if _, err := tc.Ping(cf.Context); err != nil {
+		pr, err := tc.Ping(cf.Context)
+		if err != nil {
 			return nil, trace.Wrap(err)
 		}
+		tc.LoadAllHostCAs = pr.Auth.LoadAllHostCAs
+
+		if tc.LoadAllHostCAs {
```

Comment:
Thinking about this more - if the user provided identity file, I don't think we should load stuff from local keystore at all? Everything we need should be in the identity file, no? cc @jakule